x-pack/plugins/ml/server/shared_services/providers/results_service.ts

```diff
@@ -25,12 +26,21 @@ export function getResultsServiceProvider({
 }: SharedServicesChecks): ResultsServiceProvider {
   return {
     resultsServiceProvider(callAsCurrentUser: LegacyAPICaller, request: KibanaRequest) {
-      const hasMlCapabilities = getHasMlCapabilities(request);
+      let hasMlCapabilities: HasMlCapabilities;
+      //  Uptime is using this service in anomaly alert, kibana alerting doesn't provide request object
+      // So we are adding a dummy request for now
+      // TODO: Remove this once kibana alerting provides request object
+      if (request.params !== 'DummyKibanaRequest') {
```

Comment:
creating an obvious dummy request like this is safer for non alerting calls. i like it as a temporary measure.
one suggestion for less code churn, it could be structured like this:
```
  const hasMlCapabilities = (request.params !== 'DummyKibanaRequest')? getHasMlCapabilities(request): (caps: string[]) =>Promise.resolve());
```

Reply:
@jgowdyelastic i made the change.